sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/pkversion/IncrementalChangeFeedProcessorImpl.java

```diff
@@ -87,8 +87,8 @@ public class IncrementalChangeFeedProcessorImpl implements ChangeFeedProcessor,
     private final ChangeFeedContextClient feedContextClient;
     private final ChangeFeedProcessorOptions changeFeedProcessorOptions;
     private final ChangeFeedObserverFactory<JsonNode> observerFactory;
-    private volatile String databaseResourceId;
-    private volatile String collectionResourceId;
+    private volatile String databaseId;
```

Comment:
Let's discuss in the design review - but this IMO is a bug. You need resource ids to identify deleted + recreated collections.

Reply:
yes it is  -> the question will be whether we want to change in the old change feed processor version. And when we change, migration will need to happen again

but on a further thought, this bug actually could very well blocking the migration story in the new lease store, because how we gonna know we are migration the correct ones